Proton/Sources/ObjC/PRTextStorage.m

```diff
@@ -138,20 +137,20 @@ - (void)setAttributes:(NSDictionary<NSString *, id> *)attrs range:(NSRange)range
     [self endEditing];
 }
 
-- (void)insertAttachmentInRange:(NSRange)range attachment:(Attachment *_Nonnull)attachment {
-    NSCharacterSet *spacerCharacterSet = attachment.spacerCharacterSet;
-    BOOL hasPrevSpacer = NO;
-    if (range.length + range.location > 0) {
-        NSUInteger characterIndex = range.location == 0 ? 0 : range.location - 1;
-        hasPrevSpacer = [spacerCharacterSet characterIsMember:[self.string characterAtIndex:characterIndex]];
-    }
+- (void)insertAttachmentInRange:(NSRange)range attachment:(NSTextAttachment *_Nonnull)attachment withSpacer: (NSAttributedString *) spacer {
+    NSCharacterSet *spacerCharacterSet = [NSCharacterSet whitespaceCharacterSet];  //attachment.spacerCharacterSet;
     BOOL hasNextSpacer = NO;
     if (range.location + 1 < self.length) {
         NSUInteger characterIndex = range.location + 1;
         hasNextSpacer = [spacerCharacterSet characterIsMember:[self.string characterAtIndex:characterIndex]];
     }
 
-    NSAttributedString *attachmentString = [attachment stringWithSpacersWithAppendPrev:!hasPrevSpacer appendNext:!hasNextSpacer];
+    NSMutableAttributedString *attachmentString = [[NSMutableAttributedString attributedStringWithAttachment:attachment] mutableCopy];
+
+    if (hasNextSpacer == NO) {
```

Comment:
```suggestion
    if (!hasNextSpacer) {
```

Reply:
I prefer not using ! operator and be more explicit purely for readability i.e. don't want to accidentally miss ! operator when glancing at the code :)